internal/server/iigointernal/legislature.go

```diff
@@ -0,0 +1,166 @@
+package iigointernal
+
+import (
+	"github.com/SOMAS2020/SOMAS2020/internal/common/baseclient"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+	"github.com/pkg/errors"
+)
+
+type legislature struct {
+	SpeakerID     shared.ClientID
+	budget        shared.Resources
+	judgeSalary   shared.Resources
+	ruleToVote    string
+	ballotBox     voting.BallotBox
+	votingResult  bool
+	clientSpeaker roles.Speaker
+}
+
+// returnJudgeSalary returns the salary to the common pool.
+func (l *legislature) returnJudgeSalary() shared.Resources {
+	x := l.judgeSalary
+	l.judgeSalary = 0
+	return x
+}
+
+// withdrawJudgeSalary withdraws the salary for the Judge from the common pool.
+func (l *legislature) withdrawJudgeSalary(gameState *gamestate.GameState) error {
+	var judgeSalary = shared.Resources(rules.VariableMap["judgeSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(judgeSalary, gameState)
+	if withdrawError != nil {
+		l.judgeSalary = judgeSalary
+	}
```

Comment:
? if it's not nil, return the error right, then the error should be handled.

Reply:
Sorry, this should be == not != . Fixed now!